ffi/derive/src/derive.rs

```diff
@@ -0,0 +1,225 @@
+use std::collections::HashSet;
+
+use proc_macro2::TokenStream;
+use proc_macro_error::{abort, OptionExt};
+use quote::quote;
+use syn::{parse_quote, visit_mut::VisitMut, Ident, ItemStruct, Type};
+
+use crate::{get_ident, impl_visitor::SelfResolver};
+
+/// What does structure derives for public fields
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+enum Derive {
+    Set,
+    Get,
+    GetMut,
+}
+
+/// Generate FFI function equivalents of derived methods
+pub fn gen_fns_from_derives(item: &ItemStruct) -> Vec<syn::ItemFn> {
+    let struct_derives = parse_derives(&item.attrs).unwrap_or_default();
+
+    let mut ffi_derives = vec![];
```

Comment:
```suggestion
    let mut ffi_derives = Vec::new();
```

Reply:
I don't think there is any difference. Do you think it's a bit faster to compile if it's not a macro expansion?

Reply:
I doubt it adds more than a few microseconds to the compile time. it's more about communicating the  intention.